src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -2028,6 +2047,40 @@ TEST_F(ApplicationManagerImplTest, AddAndRemoveQueryAppDevice_SUCCESS) {
   EXPECT_FALSE(app_manager_impl_->IsAppsQueriedFrom(device_handle));
 }
 
+TEST_F(
+    ApplicationManagerImplTest,
+    RequestForInterfacesAvailability_AllRequestsWillBeSuccessfullyRequested) {
+  std::vector<hmi_apis::FunctionID::eType> expected_requests{
+      hmi_apis::FunctionID::VehicleInfo_IsReady,
+      hmi_apis::FunctionID::VR_IsReady,
+      hmi_apis::FunctionID::TTS_IsReady,
+      hmi_apis::FunctionID::UI_IsReady,
+      hmi_apis::FunctionID::RC_IsReady};
+
+  for (auto request : expected_requests) {
+    SetUpRequestForInterfacesAvailabilityExpectation(request);
+  }
+
+  EXPECT_CALL(
+      *mock_rpc_service_,
+      ManageHMICommand(
+          HMIResultCodeIs(hmi_apis::FunctionID::VehicleInfo_IsReady), _));
+  EXPECT_CALL(
+      *mock_rpc_service_,
+      ManageHMICommand(HMIResultCodeIs(hmi_apis::FunctionID::VR_IsReady), _));
+  EXPECT_CALL(
+      *mock_rpc_service_,
+      ManageHMICommand(HMIResultCodeIs(hmi_apis::FunctionID::TTS_IsReady), _));
+  EXPECT_CALL(
+      *mock_rpc_service_,
+      ManageHMICommand(HMIResultCodeIs(hmi_apis::FunctionID::UI_IsReady), _));
+  EXPECT_CALL(
+      *mock_rpc_service_,
+      ManageHMICommand(HMIResultCodeIs(hmi_apis::FunctionID::RC_IsReady), _));
```

Comment:
@ydementieiev I think `SetUpRequestForInterfacesAvailabilityExpectation` should contain these expectations.

Reply:
@mvorobio I'm not agree,  `SetUpRequestForInterfacesAvailabilityExpectation` contain ON_CALL's for correct `RequestForInterfacesAvailability` function  work in test. But the test verifies will be sent ManageHMICommand with correct params. Because of that they need be inside test and we can see what we expect

Reply:
@ydementieiev What about `for` loop then?
```cpp
  for (auto request : expected_requests) {
    SetUpRequestForInterfacesAvailabilityExpectation(request);
    EXPECT_CALL(
      *mock_rpc_service_,
      ManageHMICommand(HMIResultCodeIs(request), _));
  }
```
Also the name `SetUpRequestForInterfacesAvailabilityExpectation` is a little bit confusing.

Reply:
@mvorobio [check now](https://github.com/LuxoftSDL/sdl_core/pull/38/commits/ea0219d6d0e6c350e8f6ec92c405c1af124b0233)